packages/arb-ts/src/lib/l1Bridge.ts

```diff
@@ -102,82 +109,60 @@ export class L1Bridge {
       .then(([res]) => res)
   }
 
-  public async getAndUpdateL1TokenData(
-    erc20L1Address: string
-  ): Promise<L1TokenData> {
-    const tokenData = this.l1Tokens[erc20L1Address] || {
-      ERC20: undefined,
-      CUSTOM: undefined,
-    }
+  public async contractExists(contractAddress: string): Promise<boolean> {
+    const contractCode = await this.l1Provider.getCode(contractAddress)
+    return !(contractCode.length > 2)
+  }
+
+  public async getL1TokenData(erc20L1Address: string): Promise<L1TokenData> {
     const walletAddress = await this.getWalletAddress()
     const gatewayAddress = await this.getGatewayAddress(erc20L1Address)
 
-    if (!tokenData.ERC20) {
-      if ((await this.l1Provider.getCode(erc20L1Address)).length > 2) {
-        // If this will throw if not an ERC20, which is what we *want*.
-        const ethERC20TokenContract = await ERC20__factory.connect(
-          erc20L1Address,
-          this.l1Signer
-        )
-        const [balance] = await ethERC20TokenContract.functions.balanceOf(
-          walletAddress
-        )
+    if (!this.contractExists(erc20L1Address))
+      throw new Error(`No ERC20 at ${erc20L1Address} `)
```

Comment:
Re: the comment below before `ERC20__factory.connect`, is this necessary?

Reply:
Actually the balanceOf call will definitely throw

Reply:
yea, not necessary, only if we wanted to surface the reason for a throw more accurately